cli/compiler.rs

```diff
@@ -164,6 +178,39 @@ fn lazy_start(parent_state: Arc<IsolateState>) -> CompilerShared {
               Ok(())
             })
           }));
+          // All worker responses are here initially before being sent via
+          // their respective sender. This system can be compared to the
+          // promise system used on the js side. It provides a way to
+          // resolve many futures via the same channel.
+          runtime.spawn(lazy(move || {
+            debug!("Start worker stream handler!");
+            let worker_stream = resources::get_message_stream_from_worker(rid);
+            worker_stream
+              .for_each(|msg: Buf| {
+                let res_json = std::str::from_utf8(&msg).unwrap();
+                debug!("Got message from worker: {}", res_json);
+                // Get the intended receiver from the message.
+                let cmd_id =
```

Comment:
Can you abstract out the below stanza into a stand alone function? (ideally with a unit test)